client/post-editor/editor-deprecation-dialog/index.jsx

```diff
@@ -0,0 +1,134 @@
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+/**
+ * WordPress dependencies
+ */
+import { Button, Modal } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import isEditorDeprecationDialogShowing from 'state/selectors/is-editor-deprecation-dialog-showing';
+import { hideEditorDeprecationDialog } from 'state/ui/editor-deprecation-dialog/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { setSelectedEditor } from 'state/selected-editor/actions';
+import { localize } from 'i18n-calypso';
+import { composeAnalytics, recordTracksEvent, withAnalytics } from 'state/analytics/actions';
+import { getEditorPostId } from 'state/ui/editor/selectors';
+import { getEditedPostValue } from 'state/posts/selectors';
+import getGutenbergEditorUrl from 'state/selectors/get-gutenberg-editor-url';
+
+/**
+ * Style dependencies
+ */
+import '@wordpress/components/build-style/style.css';
+import './style.scss';
+
+class EditorDeprecationDialog extends Component {
```

Comment:
Since this is a new component maybe it would be better to use a Functional component, so we can take advantage of hooks? On the other hand I guess we'll be deleting this in a few months so it probably doesn't matter much :)

Reply:
Eh, I'm not so concerned about the amazing-ness of code that's born to die. 😉